prov/util/src/util_attr.c

```diff
@@ -1182,15 +1185,9 @@ void ofi_alter_info(struct fi_info *info, const struct fi_info *hints,
 		 * the checks depend on unmodified provider mr_mode attr */
 		info->caps = ofi_get_info_caps(info, hints, api_version);
 
-		if (FI_VERSION_LT(api_version, FI_VERSION(1, 5))) {
-			if (info->domain_attr->mr_mode & FI_MR_LOCAL)
-				info->mode |= FI_LOCAL_MR;
-		}
```

Comment:
The reason info->mode was altered here is so that a call like alter_domain_attr() doesn't need to alter anything other than domain attributes, which ends up looking like a unexpected side effect.

Reply:
Agreed. I wanted to keep the checks in one place but I guess having the same checks here wouldn't hurt and it helps to  avoid the side effect.